app/imageUtils.js

```diff
@@ -0,0 +1,208 @@
+const fs = require('fs')
+const path = require('path')
+const sharp = require('sharp')
+const crypto = require('crypto')
+const User = require('./models/user')
+const logger = require('./logger')
+const Article = require('./models/article')
+
+const THREAD_COUNT = require('os').cpus().length
+const PROFILE = 'profile', ARTICLE = 'article'
+const USER_PROFILE_IMAGENAME_LENGTH = 12
+const ARTICLE_HEADER_IMAGENAME_LENGTH = 16, ARTICLE_HEADER_ID = 37, ARTICLE_BODY_ID = 14
+const HEADER_IMAGE_SIZES = ["400", "650", "864", "1024", "1920", "2048"]
```

Comment:
We should probably export it to some config.
Preferably we should also share this data with frontend through EJS and instead of putting sizes in two places, we'd put it in one place and the frontend will pull the data from there.

Reply:
We'll do this in a later ticket